src/test/kotlin/carracing/RacingCarsTest.kt

```diff
@@ -0,0 +1,113 @@
+package carracing
+
+import io.kotest.assertions.throwables.shouldThrowExactly
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContainExactly
+import io.kotest.matchers.shouldBe
+
+class RacingCarsTest : StringSpec({
+
+    "getMaxDistanceCars 는 distance 가 가장 큰 차들을 가져온다" {
+        // given
+        val racingCars = RacingCars(
+            listOf<Car>(
+                Car("TestCar1", 3),
+                Car("TestCar2", 4),
+                Car("TestCar3", 10),
+                Car("TestCar4", 0),
+                Car("TestCar5"),
+            )
+        )
+
+        // when
+        val maxDistanceCars = racingCars.getMaxDistanceCars()
+
+        // then
+        maxDistanceCars.size shouldBe 1
+        maxDistanceCars[0].name shouldBe "TestCar3"
+        maxDistanceCars[0].distance shouldBe 10
+    }
+
+    "getMaxDistanceCars 는 가장 큰 distance 가 같은 차가  여러개인 경우 그 차를 전부 가져온다" {
+        // given
+        val racingCars = RacingCars(
+            listOf<Car>(
+                Car("TestCar1", 3),
+                Car("TestCar2", 30),
+                Car("TestCar3", 30),
+                Car("TestCar4", 30),
+                Car("TestCar5", 10),
+                Car("TestCar6", 4),
+            )
+        )
+
+        // when
+        val maxDistanceCars = racingCars.getMaxDistanceCars()
+
+        // then
+        maxDistanceCars.size shouldBe 3
+        maxDistanceCars.map { car: Car -> car.name } shouldContainExactly listOf(
+            "TestCar2",
+            "TestCar3",
+            "TestCar4"
+        )
+        maxDistanceCars.map { car: Car -> car.distance } shouldContainExactly List(3) { 30 }
```

Comment:
이쪽에서 Class List 를 비교할수 있는 방법이 있을까요?

Reply:
저라면 아래와 같이 구현했을 것 같아요!!
```suggestion
        val expectedWinners = listOf(racingCars.cars[1], racingCars.cars[2], racingCars.cars[3])

        maxDistanceCars shouldContainExactly expectedWinners
```